From 3687a5697b87709a9d8d2b4031667d8ea6868b38 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Fri, 24 Jun 2022 09:32:08 +0300
Subject: [PATCH] [SW-91832] gaudi2: erom: add security boot_info

HL_BOOT_INFO_SECURITY_OTP was not getting set in
erom_recovery application. This was causing invalid
otp in fw loader agent which reads otp from boot_info.

In this patch code is aligned with preboot to call
preboot_security_sp_sram_init() function from erom_recovery()
app too. This function takes care to set SECURITY_OTP as
well as SECURITY_FUSE for fake security.

Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
Change-Id: I9f8f4f17974740751f11c7587224c11a6edcf737
---

diff --git a/app/ext_rom/erom_recovery.c b/app/ext_rom/erom_recovery.c
index a79666a..ea25bcc 100644
--- a/app/ext_rom/erom_recovery.c
+++ b/app/ext_rom/erom_recovery.c
@@ -125,6 +125,7 @@
 	preboot_version();
 	preboot_security_dynamic_vers_init(ver_off, sizeof(preboot_vers));
 
+	preboot_security_sp_sram_init();
 	preboot_dev_enable_caps();
 
 	cpu_boot_sts_set(CPU_BOOT_STATUS_WAITING_FOR_BOOT_FIT);
diff --git a/arch/gaudi/arm/security.c b/arch/gaudi/arm/security.c
index ace3328..951bd87 100644
--- a/arch/gaudi/arm/security.c
+++ b/arch/gaudi/arm/security.c
@@ -255,7 +255,7 @@
 	boot_info_set(HL_BOOT_INFO_SECURITY_OTP, otp, sizeof(*otp));
 }
 
-static void preboot_security_sp_sram_init(void)
+void preboot_security_sp_sram_init(void)
 {
 	preboot_security_sp_sram_write_otp();
 	preboot_security_sp_sram_write_efuse();
diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 93cbb68..718d9b5 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -493,7 +493,7 @@
 	boot_info_set(HL_BOOT_INFO_KEY_IDX, key_idx, sizeof(u32));
 }
 
-static void preboot_security_sp_sram_init(void)
+void preboot_security_sp_sram_init(void)
 {
 	if (preboot_fake_security_is_enabled()) {
 		preboot_puts("preboot in fake security mode\n");
diff --git a/include/preboot.h b/include/preboot.h
index 8f0b300..be68257 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -256,6 +256,7 @@
 void ppboot_secure_arc(void);
 
 void hl_flr_report_err(u32 err_bit);
+void preboot_security_sp_sram_init(void);
 
 #endif /* __ASSEMBLY__ */
 
